src/stream/src/executor/partition_compute/mod.rs

```diff
@@ -0,0 +1,270 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod iceberg;
+use std::fmt::{Debug, Formatter};
+
+use futures_async_stream::try_stream;
+pub use iceberg::*;
+use risingwave_common::array::{ArrayImpl, ArrayRef, StructArray};
+use risingwave_common::catalog::Schema;
+use risingwave_pb::stream_plan::partition_compute_node::PartitionSpec;
+
+use super::{
+    ActorContextRef, BoxedExecutor, BoxedMessageStream, Executor, ExecutorInfo, Message,
+    PkIndicesRef, *,
+};
+
+pub trait TransformBuilder: Debug + Send {
+    fn build(spec: PartitionSpec) -> StreamResult<Vec<PartitionComputeField>>;
+}
+
+pub trait Transform: Debug + Send {
+    fn transform(&self, array: &ArrayRef) -> StreamResult<ArrayRef>;
+}
+
+pub struct PartitionComputeField {
+    col_index: usize,
+    transform: Box<dyn Transform>,
+}
+
+pub struct PartitionComputeExecutor {
```

Comment:
I'm wondering if we really need an extra `PartitionComputeExecutor`, this is in fact a `ProjectionExecutor`,  and what's really required is a `TransformExpression`

Reply:
Yes, I think we use a Project instead and append the partition field, like:🤔
`|field 1|field 2| -> |field 1|field 2|partition field 1|partition field 2|`

Reply:
I think it's better to put all partition fields under one struct type? This we can avoid name conflict as much as possible?

Reply:
Yes. And we should implement an expression which take multiple field and combine them as a struct first, seems we don't have a such a expression right?

Reply:
I think we can do this just in the exprssion? StructArray is also an array.

Reply:
oh, I got your point.